### PR TITLE
test: improve coverage for GET '/collections/{collection_uuid}' AND handle CXG

### PR DESCRIPTION
**Pull request by danieljhegeman**
- #TICKET_NUMBER

### Reviewers
**Functional:** 
@Bento007 
**Readability:** 
@nayib-jose-gloria 
---

## Changes
Update code and tests to ensure...
- CXG asset type is *not* returned (for `GET /collections` OR `GET /assets`)
- ontology elements are repackaged as arrays if not already

### CODE REVIEW

File: backend/config/curation-api.yml [hunk 1]
```diff
@@ -496,16 +496,14 @@ components:
...
-              presigned_url:
-                type: string
-              file_name:
-                type: string
-              file_size:
-                type: number
-              filetype:
-                type: string
+            allOf:
+              - $ref: "#/components/schemas/dataset_asset"
```
Comment (danieljhegeman) [inline backend/config/curation-api.yml:500]: @Bento007 I was surprised your asset endpoint was passing the schema validation (due to cxg). Added this so it gets the benefit of the enum def validation

File: backend/corpora/lambdas/api/v1/curation/collections/collection_uuid/assets.py [hunk 2]
```diff
@@ -20,6 +21,8 @@ def get(collection_uuid: str, curator_tag: str = None, dataset_uuid=None):
     error_flag = False
     for a in assets:
         asset = DatasetAsset(a)
+        if asset.filetype == DatasetArtifactFileType.CXG:
+            continue
```
[the remark this answers is not in the document]
Comment (Bento007) [reply backend/corpora/lambdas/api/v1/curation/collections/collection_uuid/assets.py:25]: currently that should never happen because all cxgs are created from an H5AD. If it does, it's a bug.

File: tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py [hunk 3]
```diff
@@ -182,109 +187,114 @@ def test__get_only_private_collections_with_auth__OK(self):
...
+
     def setUp(self):
         super().setUp()
         self.test_collection = dict(
             name="collection", description="description", contact_name="john doe", contact_email="johndoe@email.com"
         )
 
-    def test__get_public_collection_verify_body__OK(self):
+    def test__get_public_collection_verify_body_is_reshaped_correctly__OK(self):
+        dataset = self.session.query(DbDataset).filter(DbDataset.id == "test_dataset_id").one_or_none()
```
Comment (Bento007) [inline tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py:285]: Why do you need to change it to a dict and is this related to the PR or extra?

**Review by Bento007 — APPROVED** [review]

File: tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py (hunk 3 above)
Comment (danieljhegeman) [reply tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py:285]: I'll add more comments to explain this.

It's to make sure that the reshaping function correctly converts lone dicts to _lists_ of dicts (of length 1).

The return type needs to be an array.

**Review by nayib-jose-gloria — APPROVED** [review]
